chainstate/test-framework/src/framework.rs

```diff
@@ -82,23 +82,19 @@ impl TestFramework {
         blocks: usize,
         rng: &mut impl Rng,
     ) -> Result<Id<GenBlock>, ChainstateError> {
-        // TODO: Instead of creating TestBlockInfo on every iteration, a proper UTXO set
-        // abstraction should be used. See https://github.com/mintlayer/mintlayer-core/issues/312
-        // for the details.
-        let mut prev_block = TestBlockInfo::from_id(&self.chainstate, *parent_block);
-
+        let mut prev_block_id = *parent_block;
         for _ in 0..blocks {
             let block = self
                 .make_block_builder()
                 // .with_transactions(vec![transaction])
```

Comment:
It isn't related to you changes, but I think that we should remove the commented line.

Reply:
Nice catch Stanislav, thank you.